client/src/pages/metadata-query/__tests__/geo-query-type.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { describe, expect, test, vi } from 'vitest';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import React from 'react';
+
+import GeoQueryType from '@/pages/metadata-query/geo-query-type';
+
+describe('Test BandSelection', () => {
+
+  const mockHandle = vi.fn();
+  const selected = 'annotations';
+
+  test('Basic Rendering', () => {
+    render(
+      <GeoQueryType
+        handleSelection={mockHandle}
+        selected={selected}
+        name={'captures'}
+      />
+    );
+    const type = screen.getByLabelText('captures');
+    fireEvent(type, new MouseEvent('click', {
+      bubbles: true,
+      cancelable: true,
+    }));
+    expect(mockHandle).toHaveBeenCalledTimes(1);
+    expect(mockHandle).toHaveBeenCalledWith('captures');
+  });
```

Comment:
Pls no

Reply:
Couold you provide any reason whhy or a viable alternative ?

Reply:
How does calling a method prove the functionality we want is done. Test the return you see in the page rather than that. I.e is the button still disabled vs is it not.

Reply:
This is the same case of the other methos thic component in isolation does not provide impediate feedback for any change in this method. it only makes a callback to to object that did mount that.